plugins/ssr/ssrparser.mlg

```diff
@@ -79,8 +80,35 @@ let ssrtac_entry name = {
   mltac_index = 0;
 }
 
-let register_ssrtac name f =
-  Tacenv.register_ml_tactic (ssrtac_name name) [|f|]
+let cache_tactic_notation (_, (key, body, parule)) =
+  Tacenv.register_alias key body;
+  Pptactic.declare_notation_tactic_pprule key parule
+
+type tactic_grammar_obj = KerName.t * Tacenv.alias_tactic * Pptactic.pp_tactic
+
+let inSsrGrammar : tactic_grammar_obj -> obj =
+  declare_object {(default_object "SsrGrammar") with
+                  load_function = (fun _ -> cache_tactic_notation);
+                  cache_function = cache_tactic_notation;
+                  classify_function = (fun x -> Keep x)}
+
+let register_ssrtac name f prods =
+  let open Pptactic in
+  Tacenv.register_ml_tactic (ssrtac_name name) [|f|];
+  let map id = Reference (Locus.ArgVar (CAst.make id)) in
+  let get_id = function
+    | TacTerm s -> None
+    | TacNonTerm (_, (_, ido)) -> ido in
+  let ids = List.map_filter get_id prods in
+  let tac = TacML (CAst.make (ssrtac_entry name, List.map map ids)) in
+  let key = Lib.make_kn (Names.Id.of_string_soft ("ssreflect_plugin@#" ^ name)) in
```

Comment:
You probably don't have to generate a hacky name here.

Reply:
Well, I want to make sure not to collide with names generated by `Tacentries.make_fresh_key`. What name generation scheme do you suggest?

Reply:
It's not going to clash because the module name is going to be distinct. That said, if I am reading correctly you're adding the libobject at toplevel? If so this is wrong and may explain the error.

Reply:
So, indeed, you need to wrap all your libobject declarations inside `Mltop.declare_cache_obj`, the latter having to be called at toplevel of the plugin. You have to move a bit your code around for that, in particular you can't get the key and declare the object at the same time.

Reply:
Ah yes, I already suspected it had something to do with `declare_cache_obj`. Will give it a try.

Reply:
I´ve made an attempt, but this solution is rather convoluted in my opinion. Let me know if you can suggest some simplifications.